CMakeLists.txt

```diff
@@ -1,123 +1,115 @@
+
+# Force out-of-source build
+if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+  message(FATAL_ERROR
+    "This project requires an out of source build. Remove the file 'CMakeCache.txt' found in this directory before continuing, create a separate build directory and run 'cmake [options] <srcs>' from there."
+  )
+endif()
+
+if(TILES)
+    # Find SDL, SDL_ttf & SDL_image for graphical install
+    find_package(SDL2)
+    if(NOT SDL2_FOUND)
+        message(FATAL_ERROR
+            "This project requires SDL2 to be installed to be compiled in graphical mode. Please install the SDL2 development libraries, or try compiling without the -DTILES=1 for a text only compilation."
+        )
+    endif()
 
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.8)
+    find_package(SDL2_ttf)
+    if(NOT SDL2_TTF_FOUND)
+        message(FATAL_ERROR
+            "This project requires SDL2_ttf to be installed to be compiled in graphical mode. Please install the SDL2_ttf development libraries, or try compiling without the -DTILES=1 for a text only compilation."
+        )
+    endif()
 
-PROJECT(CataclysmDDA)
+    find_package(SDL2_image)
+    if(NOT SDL2_IMAGE_FOUND)
+        message(FATAL_ERROR
+            "This project requires SDL2_image to be installed to be compiled in graphical mode. Please install the SDL2_image development libraries, or try compiling without the -DTILES=1 for a text only compilation."
+        )
+    endif()
 
-OPTION(LOCALIZE "Toggle gettext localization/translation." ON)
-OPTION(SDL "Use SDL curses emulation instead of terminal support. Also enables tiles support." OFF)
-OPTION(SOUND "Use SDL mixer to play music and sounds." ON)
-OPTION(LUA "Enable lua scripting support." ON)
-OPTION(LUA_BINARY "Select the lua binary to execute lua scripts with." "lua")
+    add_definitions(-DTILES -DSDLTILES)
+else()
+    # Find the ncurses library for a text based compile
+    set(CURSES_NEED_NCURSES TRUE)
+    find_package(Curses)
+    if(NOT CURSES_FOUND)
+        message(FATAL_ERROR
+            "This project requires ncurses to be installed to be compiled in text only mode. Please install the ncurses development libraries, or try compiling with the -DTILES=1 for a graphical compilation."
+        )
+    endif()
+endif()
+
+if(SOUND)
+    # You need TILES to be able to use SOUND
+    if(NOT TILES)
+        message(FATAL_ERROR
+            "You must enable graphical support with -DTILES=1 to be able to enable sound support."
+        )
+    endif()
 
-# Note: The CMake documentation says it's better to list the actual source
-#       files here, as otherwise cmake won't know to rerun when a new file
-#       is added. However, currently the Makefile also just scans for source
-#       files, so this approach should be "good enough".
-FILE(GLOB CataclysmDDA_SRCS src/*.cpp)
-FILE(GLOB CataclysmDDA_HDRS src/*.h)
+    # Sound requires ogg, vorbis & vorbisfile libraries
+    find_package(OggVorbis)
```

Comment:
You're right, I just discovered that mixer for SDL2 will load those library dynamically and is not build to require them directly. Requiring them here seem appropriate.